src/CalcViewModel/Common/TraceLogger.h

```diff
@@ -13,19 +13,19 @@ static const int maxFunctionSize = (int)CalculationManager::Command::CommandBINE
 // This class implements a singleton model ensure that only one instance is created.
 namespace CalculatorApp
 {
-    struct FuncLog
+    struct ButtonLog
     {
     public:
         int count;
-        std::wstring funcName;
-        FuncLog()
+        int buttonId;
+        std::wstring buttonName;
```

Comment:
buttonId and buttonName are redundant pieces of information. Rather than storing both of them in memory, let's just get the buttonName string at the time we actually need it, in LogButtonUsage.

Then you could define an equality operator in this class to simplify the find_if in UpdateButtonUsage.